samples/stock/demo/candle-stick-volume-width/demo.js

```diff
@@ -0,0 +1,174 @@
+(function (H) {
+    function volumeBasedWidthResizer(proceed, ...args) {
+        const ret = proceed.apply(this, args),
+            series = this;
+
+        if (series.options.baseVolume && series.is('column') && series.points) {
+            const volumeSeries = series.chart.get(series.options.baseVolume);
+            const processedYData = volumeSeries.getColumn('y', true);
+            if (volumeSeries && processedYData) {
+                const maxVolume = volumeSeries.dataMax,
+                    metrics = series.getColumnMetrics(),
+                    baseWidth = metrics.width;
+
+                series.points.forEach((point, i) => {
+                    const volume = processedYData[i];
+                    const scale = volume / maxVolume;
+                    const width = baseWidth * scale;
+
+                    if (point.shapeArgs) {
+                        point.shapeArgs.x = point.shapeArgs.x - width /
+                            2 + point.shapeArgs.width / 2;
+                        point.shapeArgs.width = width;
+                    }
+                });
+            }
+        }
+
+        return ret;
+    }
+
+    H.wrap(
+        H.seriesTypes.column.prototype,
+        'translate',
+        volumeBasedWidthResizer
+    );
+
+    H.wrap(
+        H.seriesTypes.candlestick.prototype,
+        'translate',
+        volumeBasedWidthResizer
+    );
```

Comment:
I know it wasn't part of the task, but Candlestick inherits prototype from Column, so no need to wrap both `translate` methods.

While at it, could you also check if `afterColumnTranslate` event could be used here? Instead of `wrap()`?